pkg/metaserver/spd/manager.go

```diff
@@ -19,214 +19,94 @@ package spd
+	spd, err := m.fetcher.GetSPD(ctx, pod)
 	if err != nil {
-		klog.Errorf("[spd-manager] get spd targetConfig config failed: %v, use local cache instead", err)
-		targetConfig = &configapis.TargetConfig{
-			ConfigNamespace: namespace,
-			ConfigName:      name,
-		}
-		_ = s.emitter.StoreInt64(metricsNameGetCNCTargetConfigFailed, 1, metrics.MetricTypeNameCount, baseTag...)
+		return false, err
 	}
 
-	// try to update spd cache from remote if cache spd hash is not equal to target config hash,
-	// the rate of getting remote spd will be limited by spd ServiceProfileCacheTTL
-	err = s.updateSPDCacheIfNeed(ctx, originSPD, targetConfig)
+	indicatorTarget, err := util.GetServiceBusinessIndicatorTarget(spd)
 	if err != nil {
-		klog.Errorf("[spd-manager] failed update spd cache from remote: %v, use local cache instead", err)
-		_ = s.emitter.StoreInt64(metricsNameUpdateCacheFailed, 1, metrics.MetricTypeNameCount, baseTag...)
+		return false, err
 	}
 
-	// get current spd after cache updated
-	currentSPD := s.spdCache.GetSPD(key)
-	if currentSPD != nil {
-		return currentSPD, nil
-	}
-
-	_ = s.emitter.StoreInt64(metricsNameCacheNotFound, 1, metrics.MetricTypeNameCount, baseTag...)
-
-	return nil, fmt.Errorf("get spd cache for %s not found", key)
-}
-
-// getSPDTargetConfig get spd target config from cnc
-func (s *spdManager) getSPDTargetConfig(ctx context.Context, namespace, name string) (*configapis.TargetConfig, error) {
-	currentCNC, err := s.cncFetcher.GetCNC(ctx)
+	indicatorValue, err := util.GetServiceBusinessIndicatorValue(spd)
 	if err != nil {
-		return &configapis.TargetConfig{}, err
-	}
-
-	for _, target := range currentCNC.Status.ServiceProfileConfigList {
-		if target.ConfigNamespace == namespace && target.ConfigName == name {
-			return &target, nil
-		}
+		return false, err
 	}
 
-	return nil, fmt.Errorf("get target spd %s/%s not found", namespace, name)
-}
-
-// updateSPDCacheIfNeed checks if the previous spd has changed, and
-// re-get from APIServer if the previous is out-of date.
-func (s *spdManager) updateSPDCacheIfNeed(ctx context.Context, originSPD *workloadapis.ServiceProfileDescriptor,
-	targetConfig *configapis.TargetConfig) error {
-	s.mux.Lock()
-	defer s.mux.Unlock()
-
-	if originSPD == nil && targetConfig == nil {
-		return nil
-	}
-
-	now := time.Now()
-	if originSPD == nil || util.GetSPDHash(originSPD) != targetConfig.Hash {
-		key := native.GenerateNamespaceNameKey(targetConfig.ConfigNamespace, targetConfig.ConfigName)
-		if lastFetchRemoteTime := s.spdCache.GetLastFetchRemoteTime(key); lastFetchRemoteTime.Add(s.ServiceProfileCacheTTL).After(time.Now()) {
-			return nil
-		} else {
-			// first update the timestamp of the last attempt to fetch the remote spd to
-			// avoid frequent requests to the api-server in some bad situations
-			s.spdCache.SetLastFetchRemoteTime(key, now)
+	for indicatorName, target := range indicatorTarget {
+		if target.UpperBound != nil && indicatorValue[indicatorName] > *target.UpperBound {
```

Comment:
Logic here is questionable: judged as degraded when indicator value > upperbound or indicator value < lowerbound